parser/ini_test.go

```diff
@@ -0,0 +1,436 @@
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		parser.ReadFromString(content)
+		got := parser.GetSectionKeys("Profile")
+		want := []string{"name", "password"}
+
+		if !reflect.DeepEqual(want, got) {
+			t.Errorf("Got: %v. Expected: %v", got, want)
+		}
+	})
+
+	t.Run("getsection_fromfile", func(t *testing.T) {
+		parser := NewParser()
+		parser.ReadFromFile(file_path)
+		got := parser.GetSectionKeys("Owner")
+		want := []string{"name", "email"}
+		if !reflect.DeepEqual(want, got) {
+			t.Errorf("Got: %v. Expected: %v", got, want)
+		}
+	})
+
+}
+
+func TestWriteToFile(t *testing.T) {
+	t.Run("writetofile_fromstring", func(t *testing.T) {
+		parser := NewParser()
+		content := `
+		[Profile]
+		name = jarvis
+		password = secret
+		[Deployment]
+		project = peertube
+		name = peertest
+		public_ip = true
+		cpu = 4
+		memory = 8192
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		parser.ReadFromString(content)
+		got := parser.WriteToFile("./testdata/writetofile_string_sample.ini")
+		// want := true
+		_, isFile := os.Stat("./testdata/writetofile_string_sample.ini")
+
+		if got != nil {
+			t.Errorf("Got: %v.", got)
+
+		}
+		if isFile != nil {
+			t.Errorf("File wasn't created")
+
+		}
+
+	})
+
+	t.Run("writetofile_fromfile", func(t *testing.T) {
+		parser := NewParser()
+		parser.ReadFromFile(file_path)
+		got := parser.WriteToFile("./testdata/writetofile_sample.ini")
+		// want := true
+		_, isFile := os.Stat("./testdata/writetofile_sample.ini")
+
+		if got != nil {
+			t.Errorf("Got: %v. ", got)
+
+		}
+		if isFile != nil {
+			t.Errorf("File wasn't created")
+
+		}
+	})
+
+}
+
+func TestWrongValues(t *testing.T) {
+
+	t.Run("test_wrong_content_from_string", func(t *testing.T) {
+		parser := NewParser()
+		content := `
+		[Profile
+		name = jarvis
+		password = secret
+		[Deployment]
+		project = peertube
+		name = peertest
+		public_ip = true
+		cpu = 4
+		memory = 8192
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		err := parser.ReadFromString(content)
+
+		if err == nil {
+			t.Errorf("It should've returned an error: %v", err)
+
+		}
+
+	})
+
+	t.Run("test_wrong_content_from_file", func(t *testing.T) {
+		parser := NewParser()
+		err := parser.ReadFromFile(wrong_file_path)
+
+		if err == nil {
+			t.Errorf("It should've returned an error: %v", err)
+
+		}
+
+	})
+}
+
+func TestWrongSection(t *testing.T) {
+
+	t.Run("test_wrong_section_from_string", func(t *testing.T) {
+		parser := NewParser()
+		content := `
+		[Profile]
+		name = jarvis
+		password = secret
+		[Deployment]
+		project = peertube
+		name = peertest
+		public_ip = true
+		cpu = 4
+		memory = 8192
+		[Owner]
+		name = mo
+		email = mo@peertube.com
+	`
+		err := parser.ReadFromString(content)
+
+		if err != nil {
+			t.Errorf("Error: %v", err)
+
+		}
+
+		_, sectionErr := parser.GetSection("ownerr")
+
+		if sectionErr == nil {
+			t.Errorf("ownerr section exists")
+		}
+	})
+
+	t.Run("test_wrong_section_from_file", func(t *testing.T) {
```

Comment:
the test should have more explanation, because here we are testing getting a non existent section, while in the previous from file we tested the section line was malformed